.test/config/config.yaml

```
@@ -93,6 +93,9 @@ enrichment:
     # the species specified by resources -> ref -> species above
     pathway_database: "panther"
 
+report:
+  offer_excel: true
```

```suggestion
  offer_excel: false
```

Depending on the size of the input data this might cause extremely long runtimes as the excel writer in rust requires to read all of the input data into memory before one can write it out:

https://github.com/datavzrd/datavzrd/blob/556feb7b6a39745b42900d27fbcad26ee050e158/src/render/portable/mod.rs#L1944-L1981

---

Many thanks for checking the template and for the clarification!

Did you already check for better options for xlsx file writing? Have you for example seen this one:
https://docs.rs/simple-xlsx-writer/latest/simple_xlsx_writer/

At least it doesn't read everything into memory. But not sure, if it will be any quicker and if the interface works for how you parse stuff in datavzrd...

---

This is literally the one we are using right now 😄

---

Hmmm. Maybe switch to something else, then... :thinking:

But yeah, I think this switch of the default here is good enough for now. I'll add a comment warning of the performance hit when activating this...

---

> Hmmm. Maybe switch to something else, then... 🤔

I am not sure there is something that would allow something like flushing the writer every thousand records for example. But it'd great to have something like that for sure. Might even have to do with excel itself maybe?

> But yeah, I think this switch of the default here is good enough for now. I'll add a comment warning of the performance hit when activating this...

Yeah a warning when manually activated sounds good!